### PR TITLE
Fix #18096: Fix InlineValidator with anonymous inline function not wo…

### LINKED ISSUE #18096: EachValidator & InlineValidator don't work together
https://github.com/yiisoft/yii2/issues/17810 introduced a new way of using EachValidator: it uses a new instance of DynamicModel to run the validations on. This breaks the InlineValidators that'd use $this->$attribute in an inline anonymous function.

### What steps will reproduce the problem?
Have a model with an array attribute.
Create a rule in your model with EachValidator, and as a rule set an inline function, causing an InlineValidator instance to be created. In this inline anonymous function, try checking if the members of the array are valid by accessing $this->$attribute.

### What is the expected result?
$this should be the actual object on which the function was called, $this->$attribute should be one member of the array in each call.

### What do you get instead?
$this is bound to the original Model class, and $this->$attribute is the array attribute.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.35
| PHP version      | 7.4
| Operating system | Windows

### PR DESCRIPTION
**Pull request by trombipeti**
…rking well from EachValidator

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #18096

### CODE REVIEW

**Comment by samdark** [comment]
Seems to break tests on pre PHP 7.4.

**Comment by trombipeti** [comment]
Sorry, fixed.

File: framework/validators/InlineValidator.php [hunk 1]
```diff
@@ -81,6 +84,8 @@ public function clientValidateAttribute($model, $attribute, $view)
             $method = $this->clientValidate;
             if (is_string($method)) {
                 $method = [$model, $method];
+            } else if ($method instanceof \Closure) {
```
Comment (bizley) [inline framework/validators/InlineValidator.php:87]:
```suggestion
            } elseif ($method instanceof \Closure) {
```

File: tests/data/validators/models/ValidatorTestEachAndInlineMethodModel.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+use yii\base\Model;
+
+class ValidatorTestEachAndInlineMethodModel extends Model
+{
+    public $arrayProperty = [true, false];
+
+    public function rules()
+    {
+        return [
+            ['arrayProperty', 'each', 'rule' => [function($attribute, $params, $validator) {
```
Comment (bizley) [inline tests/data/validators/models/ValidatorTestEachAndInlineMethodModel.php:19]:
```suggestion
            ['arrayProperty', 'each', 'rule' => [function ($attribute, $params, $validator) {
```

File: tests/data/validators/models/ValidatorTestEachAndInlineMethodModel.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                if(is_array($this->$attribute)) {
```
Comment (bizley) [inline tests/data/validators/models/ValidatorTestEachAndInlineMethodModel.php:20]:
```suggestion
                if (is_array($this->$attribute)) {
```

**Review by bizley — CHANGES_REQUESTED** [review]
Some code style changes only. The rest looks good

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Comment by brandonkelly** [comment]
This change introduced a change in behavior that should be documented: #18175

**Pull request merged**